legend-sdlc-test-utils/src/main/java/org/finos/legend/sdlc/test/junit/LegendSDLCTestSuiteBuilder.java

```diff
@@ -0,0 +1,235 @@
+import org.finos.legend.engine.protocol.pure.v1.model.context.PureModelContextData;
+import org.finos.legend.pure.generated.Root_meta_pure_router_extension_RouterExtension;
+import org.finos.legend.sdlc.domain.model.entity.Entity;
+import org.finos.legend.sdlc.language.pure.compiler.toPureGraph.PureModelBuilder;
+import org.finos.legend.sdlc.serialization.EntityLoader;
+import org.finos.legend.sdlc.test.junit.pure.v1.DeprecatedServiceTestCase;
+import org.finos.legend.sdlc.test.junit.pure.v1.MappingTestCase;
+import org.finos.legend.sdlc.test.junit.pure.v1.ServiceTestCase;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.ServiceLoader;
+import java.util.Set;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+public class LegendSDLCTestSuiteBuilder
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(LegendSDLCTestSuiteBuilder.class);
+
+    private final String pureVersion;
+
+    public LegendSDLCTestSuiteBuilder(String pureVersion)
+    {
+        this.pureVersion = pureVersion;
+    }
+
+    public LegendSDLCTestSuiteBuilder()
+    {
+        this(null);
+    }
+
+    public static TestSuite buildTestSuite(String name, Path... entitiesDirectories)
+    {
+        return new LegendSDLCTestSuiteBuilder().buildSuite(name, entitiesDirectories);
+    }
+
+    public static TestSuite buildTestSuite(String name, EntityLoader entityLoader)
+    {
+        return new LegendSDLCTestSuiteBuilder().buildSuite(name, entityLoader);
+    }
+
+    public TestSuite buildSuite(String name, Path... entitiesDirectories)
+    {
+        try (EntityLoader entityLoader = EntityLoader.newEntityLoader(entitiesDirectories))
+        {
+            return buildSuite(name, entityLoader);
+        }
+        catch (RuntimeException e)
+        {
+            throw e;
+        }
+        catch (Exception e)
+        {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public TestSuite buildSuite(String name, EntityLoader entityLoader)
+    {
+        return buildSuite(name, entityLoader, null);
+    }
+
+    public TestSuite buildSuite(String name, EntityLoader entityLoader, ClassLoader classLoader)
+    {
+        return buildSuite(name, entityLoader.getAllEntities().collect(Collectors.toList()), classLoader);
+    }
+
+    public TestSuite buildSuite(String name, Collection<? extends Entity> entitiesForTesting, ClassLoader classLoader)
+    {
+        Map<String, TestSuiteBuilder> testSuiteBuilders = getTestSuiteBuilderByTypeMap(MappingTestCase.class, ServiceTestCase.class, DeprecatedServiceTestCase.class);
```

Comment:
Let's replace this with a ServiceLoader approach. Then we can eliminate the need for the deprecated service test case here.